oauth2client/file.py

```diff
@@ -20,17 +20,28 @@
 
 import os
 import threading
+import warnings
 
 from oauth2client import client
 
 
 __author__ = 'jcgregorio@google.com (Joe Gregorio)'
 
+_SYM_LINK_MESSAGE = 'File: {0}: Is a symbolic link.'
+
+_IS_DIR_MESSAGE = '{0}: Is a directory'
+
+_MISSING_FILE_MESSAGE = 'Cannot access {0}: No such file or directory'
+
 
 class CredentialsFileSymbolicLinkError(Exception):
     """Credentials files must not be symbolic links."""
 
 
+class CredentialsFileDirectoryError(Exception):
```

Comment:
Both of these may be end-user bugs.  If they specify a directory instead of a file, or if they try to get credentials from a file that doesn't exist.  I can see the argument of raising `IOError` in both cases.

I suppose it depends on how you want to define the specification of `Storage.get()`.  Right now, the specification of `Storage.get` says that it only returns `Credentials`; nothing about any exceptions or `None`.  It isn't my decision on which way to go.

If we go the exceptions route, we could have a base exception, like `client.NoCredentialsFound`, and change `Storage.get` to analyze the return value of `locked_get()` and raise an exception if it's `None`.  For the cases in this PR, we could an exception that subclasses `NoCredentialsFound` and `IOError`.  This way any existing classes that extend `Storage` can return `None` (which will raise `NoCredentialsFound` in `Storage.get()`) or raise their own exception that extends `NoCredentialsFound`.

Or we could just go with the `None` route and change any exceptions to `return None` and leave it up to the end-user to investigate.  In that case, we could still issue warnings.